cookbook/security/impersonating_user.rst

```diff
@@ -151,3 +151,49 @@ setting:
                 ),
             ),
         ));
+        
+Events
+------
+
+The firewall dispatches the ``security.switch_user`` event right after the impersonation
+completed. The ``SwitchUserEvent`` is passed to the listener, based on which you are able
+to get the target user you impersonate.
+
+The cookbook article about :doc:`Making the Locale "Sticky" during a User's Session
+<cookbook/session/locale_sticky_session>` does not take any changing locale into account. 
```

Comment:
By the way, can you please move the `doc` role on its own line instead of splitting it across two lines?